example/m4/configure.m4

```diff
@@ -7,6 +7,22 @@ AC_ARG_ENABLE([test-example],
     [test_example=yes])
 AM_CONDITIONAL([test_example], [test x$test_example = xyes ])
 
+code_instrumentation=no
+AC_ARG_ENABLE([code-instrum],
+    [  --enable-code-instrum   enable code instrumentation support],
+    [if test x$enableval = xyes; then
+        code_instrumentation=yes
```

Comment:
Use `PKG_CHECK_MODULES([PAPI], [papi-5], [code_instrumentation=yes])` here.

Reply:
You mean, using something like this?

# User directory installation
./configure PKG_CONFIG_PATH=/home/<my_papi_path>/lib/pkgconfig

# ... OR directly specifying flags
./configure PAPI_CFLAGS="-I/home/<my_papi_path>/include" PAPI_LIBS="-L/home/<my_papi_path>/lib -lpapi"

Reply:
Yes. Moreover if you use systemwide PAPI, you won't have to specify anything, just --enable-code-instrum

Reply:
--enable-code-instrum will remain, as code instrumentation is disabled by default, even if papi is available. --with-papi-path will be removed in favor  of PKG_CONFIG_PATH.

Reply:
Yes, this is fine. I also suggest to merge `--with-code-instrum-profile` with `--enable-code-instrum`.